### PR TITLE
 CLOUDP-83092: Add release task for init containers

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

~To be reviewed after: #432~

This PR adds a release task for the version upgrade post start hook and readiness probe.

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

File: pipeline.py [hunk 1]
```diff
@@ -27,12 +27,16 @@ def build_agent_image_ubi(config: DevConfig) -> None:
     args = {
         "agent_version": release["agent"]["version"],
         "tools_version": release["agent"]["tools_version"],
-        "tools_distro": "ubuntu1604-x86_64",
-        "agent_distro": "linux_x86_64",
+        "tools_distro": "rhel70-x86_64",
+        "agent_distro": "rhel7_x86_64",
```
Comment (chatton) [inline pipeline.py:31]: the values of these fields were in the opposite distro!

**Review by rodrigovalin — COMMENTED** [review]
I have one comment on that `ensure_skip_tag` function whose return value seems a bit odd.

File: scripts/dev/dev_config.py [hunk 2]
```diff
@@ -64,6 +73,12 @@ def agent_image(self) -> str:
             return self._config["agent_image_ubi"]
         return self._config["agent_image_ubuntu"]
 
+    def ensure_skip_tag(self, tag: str) -> bool:
```
[the remark this answers is not in the document]
Comment (chatton) [reply scripts/dev/dev_config.py:76]: it was intended to return true if the value was modified - but you're right we don't actually need this! Will remove

File: scripts/ci/build_and_push_image_sonar.sh [hunk 3]
```diff
@@ -7,4 +7,10 @@ set +x
 echo "${quay_password:?}" | docker login "-u=${quay_user_name:?}" quay.io --password-stdin
 
 # shellcheck disable=SC2154
-python3 pipeline.py --image-name "${image_name}"
+if [ -n "${release}" ]; then
```
[the remark this answers is not in the document]
Comment (chatton) [reply scripts/ci/build_and_push_image_sonar.sh:10]: yes that's a lot cleaner!

**Review by rodrigovalin — APPROVED** [review]
LGTM

**Pull request merged**